src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQueryAvroToStructuredTransformer.java

```diff
@@ -36,6 +36,11 @@
  * Create StructuredRecords from GenericRecords. Contains custom logic for BigQuery date and time types.
  */
 public class BigQueryAvroToStructuredTransformer extends RecordConverter<GenericRecord, StructuredRecord> {
+
+  public StructuredRecord transform(GenericRecord genericRecord) throws IOException {
+    return transform(genericRecord, Schema.parseJson(genericRecord.getSchema().toString()));
```

Comment:
parsing schema is an expensive operation - we should avoid doing it for every single record.

Since we know the schema will always be the same for all records, we can have a Schema variable that gets set for the very first record so that we only parse once.

Reply:
Done.